rebuild/src/public_key.cc

```diff
@@ -0,0 +1,154 @@
+#include "public_key.h"
+
+void PublicKey::Init(
+    Napi::Env env, Napi::Object &exports, BlstTsAddon *module) {
+    Napi::HandleScope scope(env);
+    auto proto = {
+        StaticMethod(
+            "deserialize",
+            &PublicKey::Deserialize,
+            static_cast<napi_property_attributes>(
+                napi_static | napi_enumerable)),
+        InstanceMethod(
+            "serialize",
+            &PublicKey::Serialize,
+            static_cast<napi_property_attributes>(napi_enumerable)),
+        InstanceMethod(
+            "keyValidate",
+            &PublicKey::KeyValidate,
+            static_cast<napi_property_attributes>(napi_enumerable)),
+        InstanceMethod(
+            "isInfinity",
+            &PublicKey::IsInfinity,
+            static_cast<napi_property_attributes>(napi_enumerable)),
+        InstanceMethod(
+            "multiplyBy",
+            &PublicKey::MultiplyBy,
+            static_cast<napi_property_attributes>(napi_enumerable)),
+    };
+
+    Napi::Function ctr = DefineClass(env, "PublicKey", proto, module);
+    module->_public_key_ctr = Napi::Persistent(ctr);
+    exports.Set(Napi::String::New(env, "PublicKey"), ctr);
+
+    Napi::Object js_exports = exports.Get("BLST_CONSTANTS").As<Napi::Object>();
+    js_exports.Set(
+        Napi::String::New(env, "PUBLIC_KEY_LENGTH_COMPRESSED"),
+        Napi::Number::New(env, BLST_TS_PUBLIC_KEY_LENGTH_COMPRESSED));
+    js_exports.Set(
+        Napi::String::New(env, "PUBLIC_KEY_LENGTH_UNCOMPRESSED"),
+        Napi::Number::New(env, BLST_TS_PUBLIC_KEY_LENGTH_UNCOMPRESSED));
+}
+
+Napi::Value PublicKey::Deserialize(const Napi::CallbackInfo &info) {
+    BLST_TS_FUNCTION_PREAMBLE(info, env, module)
+    Napi::Value pk_bytes_val = info[0];
+    BLST_TS_UNWRAP_UINT_8_ARRAY(pk_bytes_val, pk_bytes, "pkBytes")
+
+    std::string err_out{"BLST_ERROR: pkBytes"};
```

Comment:
I'd refactor this a bit. You are allocating space on a heap to construct an error message, even though in most cases (I assume most deserialization attempts are successful) this will not fail.

If you want to return values, probably using an `std::optional<std::string>>` or `std::unique_ptr<std::string>>` might make sense, then you'd have something along the lines:
```c++
if (auto ret = is_valid_length(...)) {
 Napi::TypeError::New(env, "BLST_ERROR: pkBytes" + *ret ).ThrowAsJavaScriptException();
 return env.Undefined();
}
```
Alternatively, you might go with throwing exceptions, which may ultimately simplify the code both here and in other places in this function.

Reply:
updated in 3028b07f0ebe8bd407494981ec6c9bcd79092598